src/features/lightspeed/vue/views/fileOperations.ts

```diff
@@ -0,0 +1,159 @@
+import * as vscode from "vscode";
+import {
+  Uri,
+  workspace,
+  FileSystemError,
+  ViewColumn,
+  window,
+  commands,
+} from "vscode";
+import { expandPath } from "../../../contentCreator/utils";
+import { CollectionFinder, AnsibleCollection } from "../../utils/scanner";
+
+export async function openNewPlaybookEditor(content: string) {
+  const options = {
+    language: "ansible",
+    content: content,
+  };
+
+  const doc = await workspace.openTextDocument(options);
+  await window.showTextDocument(doc, ViewColumn.Active);
+}
+
+export async function getCollectionsFromWorkspace(): Promise<
+  AnsibleCollection[]
+> {
+  const workspaceFolders = workspace.workspaceFolders;
+
+  if (!workspaceFolders) {
+    return [];
+  }
+  const workspaceDirectories = workspaceFolders.map((f) => f.uri.fsPath);
+  const collectionFinder = new CollectionFinder(workspaceDirectories);
+  await collectionFinder.refreshCache();
+  return collectionFinder.cache;
+}
+
+export async function getRoleBaseDir(
+  collectionName: string,
+  roleName: string,
+): Promise<Uri> {
+  const collectionFound = await getCollectionsFromWorkspace();
+  const collectionMatch = collectionFound.filter(
+    (e) => e.fqcn === collectionName,
+  );
+  if (collectionMatch.length === 0) {
+    throw new Error("Collection not found in the workspace!");
+  } else if (collectionMatch.length !== 1) {
+    throw new Error(
+      `Too many directories found for collection ${collectionName}!`,
+    );
+  }
+  const roleBaseDirUri = Uri.file(
+    `${collectionMatch[0].path}/roles/${roleName}`,
+  );
+  return roleBaseDirUri;
+}
+
+export async function fileExists(uri: Uri): Promise<boolean> {
+  try {
+    await workspace.fs.stat(uri);
+  } catch (e) {
+    if (e instanceof FileSystemError && e.code === "FileNotFound") {
+      return false;
+    }
+  }
+  return true;
+}
+
+export class FileOperations {
+  public async openLogFile(fileUrl: string) {
+    const logFileUrl = vscode.Uri.file(expandPath(fileUrl)).fsPath;
+    const parsedUrl = vscode.Uri.parse(`vscode://file${logFileUrl}`);
+    this.openFileInEditor(parsedUrl.toString());
+  }
+
+  public openFileInEditor(fileUrl: string) {
+    const updatedUrl = expandPath(String(fileUrl));
+    vscode.commands.executeCommand("vscode.open", vscode.Uri.parse(updatedUrl));
+  }
+
+  public async openRoleFolderInWorkspace(folderUrl: string, roleName: string) {
```

Comment:
Name scheme should match the other "open folder" methods, e.g. "openFolderInWorkspaceRole". Or the plugin method could be changed to "openPluginFolderInWorkspace". Either, but the scheme should match.